userdocs/src/usage/eks-private-cluster.md

```diff
@@ -0,0 +1,147 @@
+# EKS Fully-Private Cluster
+
+eksctl supports creation of fully-private clusters that have no outbound internet access and have only private subnets.
+VPC endpoints are used to enable private access to AWS services.
+
+This guide describes how to create a private cluster without outbound internet access.
+
+
+## Creating a fully-private cluster
+
+The only required field to create a fully-private cluster is `privateCluster.enabled`:
+
+```yaml
+privateCluster:
+  enabled: true
+```
+
+## Configuring private access to additional AWS services
+
+To enable worker nodes to access AWS services privately, eksctl creates VPC endpoints for the following services:
+
+- Interface endpoints for ECR (both `ecr.api` and `ecr.dkr`) to pull container images (AWS CNI plugin etc)
+- A gateway endpoint for S3 to pull the actual image layers
+- An interface endpoint for EC2 required by the `aws-cloud-provider` integration
+- An interface endpoint for STS to support Fargate and IAM Roles for Services Accounts (IRSA)
+- An interface endpoint for CloudWatch logging (`logs`) if CloudWatch logging is enabled
+
+These VPC endpoints are essential for a functional private cluster, and as such, eksctl does not support configuring or
+disabling them. However, a cluster might need private access to other AWS services (e.g., Autoscaling required by the Cluster Autoscaler).
+These services can be specified in `privateCluster.additionalEndpointServices`, which instructs eksctl to create a VPC endpoint
+for each of them.
+
+
+For example, to allow private access to Autoscaling and CloudWatch logging:
+
+```yaml
+privateCluster:
+  enabled: true
+  additionalEndpointServices:
+  # For Cluster Autoscaler
+  - "autoscaling"
+  # CloudWatch logging
+  - "log"
+```
+
+Currently, only `autoscaling` and `logs` are supported in `additionalEndpointServices`.
+
+## Nodegroups
+Only private nodegroups (both managed and self-managed) are supported in a fully-private cluster because the cluster's VPC is created without
+any public subnets. The `privateNetworking` field (`nodeGroup[*].privateNetworking` and `managedNodeGroup[*].privateNetworking`) must be
+explicitly set. It is an error to leave `privateNetworking` unset in a fully-private cluster.
+
+
+ ```yaml
+nodeGroups:
+- name: ng1
+  instanceType: m5.large
+  desiredCapacity: 2
+  # privateNetworking must be explicitly set for a fully-private cluster
+  # Rather than defaulting this field to `true` for a fully-private cluster, we require users to explicitly set it
+  # to make the behaviour explicit and avoid confusion.
+  privateNetworking: true
+
+managedNodeGroups:
+- name: m1
+  instanceType: m5.large
+  desiredCapacity: 2
+  privateNetworking: true
+
+```
+
+## Cluster Endpoint Access
+A fully-private cluster does not support modifying `clusterEndpointAccess` during cluster creation.
```

Comment:
Is there a check in commands like `eksctl update/upgrade cluster` and `eksctl utils update-cluster-endpoints` that throw an error if this is set?

Reply:
There's no check in `eksctl utils update-cluster-endpoints`. I have left it like that because I think if users still want to change the endpoint access, it should be possible. 

Why do we need a check in `eksctl updqate/upgrade cluster`?

Reply:
in `upgrade cluster` eksctl reconstructs the stacks and looks for differences before applying changes so I am not sure what will happen if someone changes these values and then performs an upgrade. Have you checked?

Reply:
The `clusterEndpointAccess` is set via an API call and not via CloudFormation. Also, the changes that are applied in `upgrade cluster` are only for adding new resources like the resources for supporting managed nodegroups.